exdb/emails.py

```diff
@@ -0,0 +1,181 @@
+
+    def send(self, *args, **kwargs):
+        if self.is_time_to_send():
+
+            emails = []
+            from_email = settings.SERVER_EMAIL
+            subject = settings.EMAIL_SUBJECT_PREFIX + ' Daily Digest'
+            for user in self.get_addrs():
+
+                # Find all the approvals that need evaluation and all the experiences that
+                # need to be approved by the current user
+                experience_approvals = ExperienceApproval.objects.filter(
+                    approver=user, experience__end_datetime__lt=now(), experience__status='ad')
+                experiences = Experience.objects.filter(Q(next_approver=user, status='pe') | Q(
+                    pk__in=experience_approvals.values('experience')))
+                experience_count = len(experiences)
+
+                status_to_display = [_('Pending Approval'), _('Needs Evaluation')]
+                experience_dict = OrderedDict()
+                for status in status_to_display:
+                    experience_dict[status] = []
+                for experience in experiences:
+                    if experience.needs_evaluation():
+                        experience_dict[_('Needs Evaluation')].append(experience)
+                    else:
+                        experience_dict[_('Pending Approval')].append(experience)
+
+                html = render_to_string('exdb/emails/daily.html',
+                                        {'experience_dict': experience_dict,
+                                         'experience_count': experience_count,
+                                         'url_prefix': settings.URL_PREFIX})
+                text = strip_tags(html)
+                recipients = (user.email,)
+                emails.append((subject, text, html, from_email, recipients))
+            send_mass_mail(emails)
+            return len(emails)
+        return 0
+
+
+class ExperienceStatusUpdate(EmailTaskBase):
+    """
+    This email should be sent to any user who has an experience they
+    authored that has changed in status to either approved or denied.
+    """
+    task_name = "Status Update"
+
+    def get_experiences(self):
+        return Experience.objects.filter(status__in=('de', 'ad'), needs_author_email=True)
+
+    def send(self, *args, **kwargs):
+        emails = []
+        from_email = settings.SERVER_EMAIL
+        subject = settings.EMAIL_SUBJECT_PREFIX + ' Experience status updated'
+        for experience in self.get_experiences():
+
+            html = render_to_string('exdb/emails/status_change.html',
+                                    {'experience': experience,
+                                     'url_prefix': settings.URL_PREFIX})
+            text = strip_tags(html)
+            recipients = (experience.author.email,)
+            emails.append((subject, text, html, from_email, recipients))
+
+            # Reset the bool so we do not send it out again
+            experience.needs_author_email = False
```

Comment:
We should use transaction.atomic here for speed as well as accuracy in the case of errors.